pkg/pool-weighted/test/ManagedPoolSettings.test.ts

```diff
@@ -641,72 +641,56 @@ describe('ManagedPoolSettings', function () {
             });
           });
 
-          // We don't run this test when an ongoing swap fee change is in progress as we can't guarantee the prior condition
-          if (!ongoingSwapFeeChange) {
-            context('when the starting swap fee is equal to the current swap fee', () => {
-              sharedBeforeEach(async () => {
-                expect(await pool.getSwapFeePercentage()).to.equal(START_SWAP_FEE);
-              });
-
-              it('does not emit a SwapFeePercentageChanged event', async () => {
-                const tx = await pool.updateSwapFeeGradually(caller, startTime, endTime, START_SWAP_FEE, END_SWAP_FEE);
-                expectEvent.notEmitted(await tx.wait(), 'SwapFeePercentageChanged');
-              });
-            });
-          }
-
           context('when the starting swap fee is different from the current swap fee', () => {
+            const UPDATE_FEE_START = START_SWAP_FEE.add(MAX_SWAP_FEE).div(2);
             sharedBeforeEach(async () => {
               const blockTimestamp = (await currentTimestamp()).add(1);
               await setNextBlockTimestamp(blockTimestamp);
-              await pool.updateSwapFeeGradually(caller, blockTimestamp, blockTimestamp, MAX_SWAP_FEE, MAX_SWAP_FEE);
-              expect(await pool.getSwapFeePercentage()).to.not.equal(START_SWAP_FEE);
+              await pool.updateSwapFeeGradually(
+                caller,
+                blockTimestamp,
+                blockTimestamp.add(DAY),
+                UPDATE_FEE_START,
+                MIN_SWAP_FEE
+              );
+            });
+
+            it('instantly sets the swap fee with the starting value', async () => {
+              expect(await pool.getSwapFeePercentage()).to.be.equal(UPDATE_FEE_START);
             });
```

Comment:
I'm not a fan of how we're not using the defined start and end timestamps and swap fees in here as it goes against why we encapsulate these tests into a function.

I would do something like this.
```
sharedBeforeEach(async () => {
    // Enforce that the Pool's current swap fee is different from the starting swap fee.
    await pool.updateSwapFeeGradually(caller, now, endTime, MAX_SWAP_FEE, MAX_SWAP_FEE);
    expect(await pool.getSwapFeePercentage()).to.not.equal(START_SWAP_FEE);
});

it('instantly sets the swap fee with the starting value', async () => {
    await pool.updateSwapFeeGradually(caller, startTime, endTime, START_SWAP_FEE, END_SWAP_FEE);

    expect(await pool.getSwapFeePercentage()).to.be.equal(START_SWAP_FEE);
});
```

Reply:
I like it!